### PR TITLE
link to ci-operator config documentation

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

File: ci-operator/config/openshift/origin/openshift-origin-master.yaml [hunk 1]
```diff
@@ -19,21 +23,33 @@ base_images:
...
     tag: oc-rpms
+
+# in addition to these tags, your repo/branch the job is being run against will be cloned
+# into an image named "src" under a path of /go/src/github.com/<orgname>/reponame
+
+# binary build command will be run from within your "src" image to produce a "bin" image.
 binary_build_commands: make build WHAT='vendor/k8s.io/kubernetes/cmd/hyperkube'
+# build_root is a specific image that will be used as the starting point for the "src" image,
+# the repository/branch for the job will be cloned on top of this image and then it will
+# be tagged as "src".      ???????
```
Comment (bparees) [inline ci-operator/config/openshift/origin/openshift-origin-master.yaml]: @smarterclayton @stevekuznetsov accurate?

File: ci-operator/config/openshift/origin/openshift-origin-master.yaml [hunk 2]
```diff
@@ -54,6 +70,8 @@ images:
       - registry.svc.ci.openshift.org/openshift/origin-v4.0:machine-os-content
       paths: null
   to: machine-os-content
+# what does promotion accomplish?  where is it promoted to and when?
+# i also heard this is needed to ensure branching works properly?
```
Comment (bparees) [inline ci-operator/config/openshift/origin/openshift-origin-master.yaml]: @smarterclayton @stevekuznetsov help

File: ci-operator/config/openshift/origin/openshift-origin-master.yaml [hunk 3]
```diff
@@ -101,6 +119,7 @@ raw_steps:
 - rpm_image_injection_step:
     from: base-machine
     to: base-machine-with-rpms
+# what magic is this? what objects do these resource settings apply to?
```
Comment (bparees) [inline ci-operator/config/openshift/origin/openshift-origin-master.yaml]: @smarterclayton @stevekuznetsov help

File: ci-operator/config/openshift/origin/openshift-origin-master.yaml [hunk 4]
```diff
@@ -180,12 +199,18 @@ resources:
     requests:
       cpu: "3"
       memory: 8Gi
+# what image does this run against/with what content?  what does the output image get tagged as?
```
Comment (bparees) [inline ci-operator/config/openshift/origin/openshift-origin-master.yaml]: @smarterclayton @stevekuznetsov help

File: ci-operator/config/sample-config.yaml [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+    tag: oc-rpms
+
+# in addition to these tags, your repo/branch the job is being run against will be cloned
+# into an image named "src" under a path of /go/src/github.com/<orgname>/reponame
+
+# binary build command will be run from within your "src" image to produce a "bin" image.
+binary_build_commands: make build WHAT='vendor/k8s.io/kubernetes/cmd/hyperkube'
+# build_root is a specific image that will be used as the starting point for the "src" image,
+# the repository/branch for the job will be cloned on top of this image and then it will
+# be tagged as "src".      ???????
```
Comment (bparees) [inline ci-operator/config/sample-config.yaml]: @smarterclayton @stevekuznetsov accurate?

File: ci-operator/config/sample-config.yaml [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+      as:
+      - fedora:29
+      paths: null
+    machine-os-content-base:
+      as:
+      - registry.svc.ci.openshift.org/openshift/origin-v4.0:machine-os-content
+      paths: null
+  to: machine-os-content
+# what does promotion accomplish?  where is it promoted to and when?
+# i also heard this is needed to ensure branching works properly?
```
Comment (bparees) [inline ci-operator/config/sample-config.yaml]: @smarterclayton @stevekuznetsov help?

File: ci-operator/config/sample-config.yaml [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+      name: stable
+      namespace: ""
+      tag: artifacts
+- rpm_image_injection_step:
+    from: base
+    to: base-with-rpms
+- rpm_image_injection_step:
+    from: base-machine
+    to: base-machine-with-rpms
+# what magic is this? what objects do these resource settings apply to?
```
Comment (bparees) [inline ci-operator/config/sample-config.yaml]: @smarterclayton @stevekuznetsov help?

File: ci-operator/config/sample-config.yaml [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+    requests:
+      cpu: "6"
+      memory: 8Gi
+  verify:
+    limits:
+      memory: 12Gi
+    requests:
+      cpu: "3"
+      memory: 8Gi
+# what image does this run against/with what content?  what does the output image get tagged as?
```
Comment (bparees) [inline ci-operator/config/sample-config.yaml]: @smarterclayton @stevekuznetsov help?

File: ci-operator/config/sample-config.yaml [hunk 9]
```diff
@@ -0,0 +1,292 @@
...
+rpm_build_commands: make build-rpms
+
+# The imagestream name and namespace to copy the initial images from, these
+# will be copied into the temporary CI job namespace.
```
Comment (bparees) [inline ci-operator/config/sample-config.yaml]: @smarterclayton @stevekuznetsov accurate?

**Comment by bparees** [comment]
@smarterclayton @stevekuznetsov i'd like to clean up this example some more(remove all the content that isn't being used as an example/illustration), but can you give me some help w/ the first pass on wording/explanations?

And if this already exists somewhere, let me know.

**Comment by petr-muller** [comment]
@bparees Similar content exists in https://github.com/openshift/ci-tools/blob/master/CONFIGURATION.md

Not saying we shouldn't have an example co-located with the real config, just I am not sure if you knew about this,

**Comment by bparees** [comment]
> @bparees Similar content exists in https://github.com/openshift/ci-tools/blob/master/CONFIGURATION.md

thanks, that is what i was looking for and did not find, perhaps it is in the wrong location or at least should be linked from the place where people are actually editing these files?

**Comment by bparees** [comment]
@petr-muller made this a symlink to the doc you referenced, ptal

**Pull request merged**